packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1039,6 +1052,56 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   *
+   * @param snapId The id of the Snap to be updated
+   * @param newVersionRange A semver version range in which the maximum version will be chosen
+   * @returns @type {TruncatedSnap} if updated, @type {null} otherwise
+   */
+  async updateSnap(
+    snapId: SnapId,
+    newVersionRange: string = DEFAULT_REQUESTED_SNAP_VERSION,
+  ): Promise<TruncatedSnap | null> {
+    const snap = this.get(snapId);
+    if (snap === undefined) {
+      throw new Error(`Couldn't find snap ${snapId} to update`);
+    }
+
+    const newSnap = await this._fetchSnap(
+      snapId as ValidatedSnapId,
+      newVersionRange,
+    );
+    if (!gtSemver(newSnap.manifest.version, snap.version)) {
+      console.warn(
+        `Tried updating snap ${snapId} within ${newVersionRange} version range, but newer version ${snap.version} already set-up`,
+      );
+      return null;
+    }
+
+    const wasRunning = this.isRunning(snapId);
+
+    if (wasRunning) {
+      this._stopSnap(snapId);
+    }
+
+    await this._add({
+      id: snapId as ValidatedSnapId,
+      manifest: newSnap.manifest,
+      sourceCode: newSnap.sourceCode,
+      versionRange: newVersionRange,
+    });
+
+    await this.authorize(snapId);
+
+    if (wasRunning) {
+      await this._startSnap({ snapId, sourceCode: newSnap.sourceCode });
```

Comment:
What happens to requests to the snap meanwhile? Is this part of https://github.com/MetaMask/snaps-skunkworks/issues/245 🤔